### PR TITLE
Trips endpoints

### PR DESCRIPTION
**Pull request by sahithimuthyala**
- revised 4 /trips endpoints
- made 4 endpoints for /trips/users
- made 4 endpoints for /trips/location
- added some functions to Trips.py and Location.py

### CODE REVIEW

File: backend/api/models/Trips.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    posted_time = DateTimeField(required=True)
+
+    cost = StringField(required=True)
+    car = ReferenceField("Car", required=True)
+    seats_available = IntField(required=True)
+    trunk_space = StringField(required=True)
+
+    past_drivers = ListField(ReferenceField("User"))
+    past_passengers = ListField(ReferenceField("User"))
+    current_users = ListField(ReferenceField("User"))
```
Comment (uawasth2) [inline backend/api/models/Trips.py]: Why do you need these in trips.py. What purpose do they serve. How can you have past drivers for a trip? Why is it in trips? same for past_passengers. Also the name current users is unclear. Are they passengers or what?

File: backend/api/models/Trips.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from mongoengine import *
+from flask_mongoengine import Document
+from .base import db
+from api.core import Mixin
+from api.models import Users, Cars, Location
+
+
+class Trip(Document, Mixin):
+    driver = ReferenceField("User", required=True)
+    passengers = ListField(ReferenceField("User", required=True))
```
Comment (uawasth2) [inline backend/api/models/Trips.py]: Do you need to have passengers? How will this act when there are no passengers. Just lmk cause I don't know how it works.

Comment (sahithimuthyala) [reply backend/api/models/Trips.py]: Passengers should not be required, so I made that change in Trips.py.

File: backend/api/models/Trips.py (hunk 1 above)
Comment (sahithimuthyala) [reply backend/api/models/Trips.py]: These 3 should not be in Trips.py. They have been deleted.

**Pull request closed without merging**